### PR TITLE
issue #1434: add the OData-Version into the No-Content response

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

* https://github.com/OData/WebApi/issues/1434

### Description

* Add the `OData-Version` into No-Content response.

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.AspNetCore.OData/Results/ResultHelpers.cs [hunk 1]
```diff
@@ -59,5 +60,18 @@ public static void AddEntityId(HttpResponse response, Func<Uri> entityId)
                 response.Headers.Add(EntityIdHeaderName, entityId().ToString());
             }
         }
+
+        public static void AddServiceVersion(HttpResponse response, Func<string> version)
```
Comment (biaol-odata) [inline src/Microsoft.AspNetCore.OData/Results/ResultHelpers.cs]: Rename to "AddServiceVersionForNoContentResponse" to indicate that it only handle the case of NoContent response code.

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/ComplexTypeInheritance/ComplexTypeInheritanceTests.cs [hunk 2]
```diff
@@ -112,6 +112,8 @@ protected override void UpdateConfiguration(WebRouteConfiguration configuration)
                 response.StatusCode,
                 requestUri,
                 contentOfString));
+
+            Assert.Equal("4.0", response.Headers.GetValues("OData-Version").Single());
```
Comment (biaol-odata) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/ComplexTypeInheritance/ComplexTypeInheritanceTests.cs]: This is a POST request with response code 201, so it is not related to the case of 204 here, right?

File: src/Microsoft.AspNetCore.OData/Results/ResultHelpers.cs (hunk 1 above)
Comment (xuzhg) [reply src/Microsoft.AspNetCore.OData/Results/ResultHelpers.cs]: Thanks. That's same syntax of the above method “AddEntityId". 
Besides, that's a public method in an internal class. It's not mandatory to add "comments" in an internal class.

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/ComplexTypeInheritance/ComplexTypeInheritanceTests.cs (hunk 2 above)
Comment (xuzhg) [reply test/E2ETest/Microsoft.Test.E2E.AspNet.OData/ComplexTypeInheritance/ComplexTypeInheritanceTests.cs]: Thanks. That's not added from my change. That's added in the OData Formatter, for example in Core version, it's added from "https://github.com/OData/WebApi/blob/master/src/Microsoft.AspNetCore.OData/Formatter/ODataOutputFormatter.cs#L183"

I added this code to verify the header.

**Comment by xuzhg** [comment]
All tests passed. @mikepizzo @AlanWong-MS @biaol-odata

**Pull request merged**